irohad/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -5,18 +5,22 @@
 
 #include "synchronizer/impl/synchronizer_impl.hpp"
 
+#include <iterator>
 #include <utility>
+#include <vector>
```

Comment:
Are iterator and vector required? I do not see their usages
```suggestion
#include <utility>
```